### PR TITLE
Pss networktest

### PR DESCRIPTION
**Pull request by zelig**

### CODE REVIEW

File: p2p/simulations/adapters/exec.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func (e *ExecAdapter) NewNode(config *NodeConfig) (Node, error) {
 	conf.Stack.P2P.NoDiscovery = true
 	conf.Stack.P2P.NAT = nil
 	conf.Stack.NoUSB = true
-	conf.Stack.Logger = log.New("node.id", config.ID.String())
+	//conf.Stack.Logger = log.New("node.id", config.ID.String())
```
Comment (lmars) [inline p2p/simulations/adapters/exec.go:107]: This has been removed on the latest `p2p-simulations-fixes` branch, rebasing `swarm-network-rewrite` on top of that should mean this change is not necessary.

File: p2p/simulations/adapters/inproc.go [hunk 2]
```diff
@@ -314,3 +339,36 @@ func (self *SimNode) NodeInfo() *p2p.NodeInfo {
...
+// credit to @lmars & Flynn
+// https://github.com/flynn/flynn/blob/master/host/containerinit/init.go#L743-L749
+// using this in large simulations requires raising OS's max open file limit
+func socketPipe() (net.Conn, net.Conn, error) {
+	pair, err := syscall.Socketpair(syscall.AF_UNIX, syscall.SOCK_STREAM, 0)
+	if err != nil {
+		return nil, nil, err
+	}
+	nameb := make([]byte, 8)
+	_, err = rand.Read(nameb)
```
Comment (lmars) [inline p2p/simulations/adapters/inproc.go:353]: Do these really need a random name?

File: p2p/simulations/adapters/inproc.go [hunk 3]
```diff
@@ -314,3 +339,36 @@ func (self *SimNode) NodeInfo() *p2p.NodeInfo {
...
+	if err != nil {
+		return nil, nil, err
+	}
+	f1 := os.NewFile(uintptr(pair[0]), string(nameb)+".out")
+	f2 := os.NewFile(uintptr(pair[1]), string(nameb)+".in")
+	pipe1, err := net.FileConn(f1)
+	if err != nil {
+		return nil, nil, err
```
Comment (lmars) [inline p2p/simulations/adapters/inproc.go:361]: Close the pair of sockets if there is an error so they don't leak.

File: p2p/simulations/network.go [hunk 4]
```diff
@@ -666,7 +666,7 @@ func (self *Network) Load(snap *Snapshot) error {
 	}
 	for _, conn := range snap.Conns {
 		if err := self.Connect(conn.One, conn.Other); err != nil {
-			return err
+			//return err
```
Comment (lmars) [inline p2p/simulations/network.go:669]: Why is this commented?

File: swarm/network/simulations/discovery/discovery_test.go [hunk 5]
```diff
@@ -100,7 +100,8 @@ func TestDiscoverySimulationSimAdapter(t *testing.T) {
 }
 
 func testDiscoverySimulationSimAdapter(t *testing.T, nodes, conns int) {
-	testDiscoverySimulation(t, nodes, conns, adapters.NewSimAdapter(services))
+	testDiscoverySimulation(t, nodes, conns, adapters.NewSocketAdapter(services))
+	// testDiscoverySimulation(t, nodes, conns, adapters.NewSimAdapter(services))
```
Comment (lmars) [inline swarm/network/simulations/discovery/discovery_test.go:104]: Remove this commented code.

File: swarm/pss/README.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
+# Postal Service over Swarm
```
Comment (lmars) [inline swarm/pss/README.md:1]: This PR seems to include all of the pss work which has already been reviewed and merged into the `pss` branch in #123, was the intention to get it all reviewed again?

**Comment by nolash** [comment]
obsoleted by https://github.com/ethersphere/go-ethereum/pull/125

**Pull request closed without merging**